File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stk = []
    for c in s:
        if c == "(":
            stk.append(c)
        elif c == ")":
            if not stk:
                return False
            else:
                stk.pop()
        
    if len(stk) == 0:
        return True
    else:
        return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>

using namespace std;

bool solution(string s)
{
    bool answer = true;
	int n=0;
    for(int i=0; i<s.size(); i++){
        if(n<0) return false;
        else if (s[i]=='(') n+=1;
        else if (s[i]==')') n-=1;
     }
    if(n==0) return true;
    else return false;
}